src/templates/renderServiceTemplates.js

```diff
@@ -0,0 +1,28 @@
+const fs = require('fs');
+const dots = require('dot');
+const crypto = require('crypto');
+
```

Comment:
When you register a function using `asFunction` then it supposed to be a factory. Factory is a function that accepts dependencies and returns a function with logic. So this module should look like this:

```js
function renderServiceTemplatesFactory(writeServiceConfigs) {
    async function renderServiceTemplates(config, homeDirPath) {
       ...
    }

    return renderServiceTemplates;
}

module.exports = renderServiceTemplatesFactory;
```

You may find a lot of examples in this repo.

Reply:
How did you learn this the first time you tried it? I was stuck for nearly two weeks on this :( and there is nothing in the [awilix docs](https://github.com/jeffijoe/awilix#asfunction) that explains that only factory functions can be used with `asFunction`... It is working now, but I still have no idea why it works.